### ISSUE TITLE
Problem details sample returns Division_by_zero when it should return 404

### ISSUE THREAD

**Issue by Rick-Anderson** [dotnet/AspNetCore.Docs.Samples#37]
[Problem details code](https://github.com/dotnet/AspNetCore.Docs.Samples/blob/main/fundamentals/middleware/problem-details-service/Program.cs#L17) returns 

```
{"type":"https://wikipedia.org/wiki/Division_by_zero",
"title":"Wrong Input",
"status":404,
"detail":"Divison by zero is not defined."}
```
When there is no endpoint rather than just 404. 

@sammychinedu2ky  fixed that by adding [`HasPath`](https://github.com/dotnet/AspNetCore.Docs.Samples/blob/main/fundamentals/middleware/problem-details-service/Program.cs#L84-L93), but that's an error prone solution.

@brunolins16  can you suggest a better mechanism to detect when there's no matching endpoint?

**Comment by sammychinedu2ky** [comment]
ooh ok let me do that

**Comment by brunolins16** [comment]
@Rick-Anderson sorry that I missed that in the first PR but can we maybe we could do:

1.  Remove the custom middleware and Set the Feature when needed. https://github.com/dotnet/AspNetCore.Docs.Samples/blob/3ebbb8baf40ec3ee07814cdf41feac1cd7a916f8/fundamentals/middleware/problem-details-service/Program.cs#L47 

2. Have the feature always there but add a new option to the [MathErrorType], eg.: NoError and check this in the CustomizeProblemDetails. (https://github.com/dotnet/AspNetCore.Docs.Samples/blob/main/fundamentals/middleware/problem-details-service/MathErrorType.cs)

In both cases i think we could use the Get instead of GetRequiredFeature and check if not `null` here https://github.com/dotnet/AspNetCore.Docs.Samples/blob/3ebbb8baf40ec3ee07814cdf41feac1cd7a916f8/fundamentals/middleware/problem-details-service/Program.cs#L16

**Comment by Rick-Anderson** [comment]
Fixed in #42
